lib/internal/modules/esm/loader.js

```diff
@@ -44,6 +49,10 @@ const { translators } = require(
   'internal/modules/esm/translators');
 const { getOptionValue } = require('internal/options');
 
+const importAssertionTypeCache = new SafeWeakMap();
+const finalFormatCache = new SafeWeakMap();
```

Comment:
@GeoffreyBooth I heard in the loaders call stream that you didn't like the `WeakMap` approach, so open to better suggestions.